### ISSUE TITLE
Tablespace symlinks lost upon recovery

### ISSUE THREAD

**Issue by vyegorov** [wal-e/wal-e#354]
I've noticed a strange behaviour on my test server:

1. restoring from `wal-e` works just fine, and all symlinks are fine
2. at the moment I start cluster (obviously, it goes into recovery straight away), symlinks created by `wal-e` are lost and the ones from the source cluster are used.

This is very unfortunate, as it is not possible to restore 2 copies of the same DB on one server,
as second restore “spoils” the already-running instance.

A bit of digging pointed me to this commit: https://www.postgresql.org/message-id/E1YsASW-0000Aa-Ut%40gemulon.postgresql.org

It appears, that when tablespace spec is provided, `tablespace_map` file should be updated accordingly after backup had been fetched.

**Comment by fdr** [comment]
Heh. That's new. I don't know what to do about it. I don't use tablespaces, so....

**Comment by vyegorov** [comment]
`pg_basebackup` produces `tablespace_map` file with appropriate tablespaces' locations. Perhaps, `wal-e` should do the same, when mapping is provided?

**Comment by fdr** [comment]
Yeah, perhaps it should. I don't know how to reconcile the old tablespace-spec support with tablespace-map exactly, especially in a backwards compatible way, and to be honest, I don't have time to study it. I would accept a high-quality patch, though.
